src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterComponent.tsx

```diff
@@ -0,0 +1,195 @@
+
+import {
+    Button,
+    ButtonComponentType,
+    ComponentSizeType,
+    ErrorScreenNotAuthorized,
+    FeatureTitleWithInfo,
+    Icon,
+    Progressing,
+    Reload,
+    showError,
+    sortCallback,
+    URLS as CommonURLS,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { importComponentFromFELibrary } from '@Components/common'
+import { ViewType } from '@Config/constants'
+import { URLS } from '@Config/routes'
+import { ClusterEnvironmentDrawer } from '@Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer'
+import CreateCluster from '@Pages/GlobalConfigurations/ClustersAndEnvironments/CreateCluster/CreateCluster.component'
+import { CreateClusterTypeEnum } from '@Pages/GlobalConfigurations/ClustersAndEnvironments/CreateCluster/types'
+
+import { getClusterList, getEnvironmentList } from './cluster.service'
+import { ClusterMetadataTypes, ClusterProps } from './cluster.type'
+import { ClusterList } from './ClusterList'
+
+const ManageCategories = importComponentFromFELibrary('ManageCategories', null, 'function')
+const ManageCategoryButton = importComponentFromFELibrary('ManageCategoryButton', null, 'function')
+
+const ClusterComponents = ({ isSuperAdmin }: ClusterProps) => {
+    const [view, setView] = useState(ViewType.LOADING)
+    const [clusters, setClusters] = useState<ClusterMetadataTypes[]>([])
+
+    const history = useHistory()
+
+    const initialize = () => {
+        Promise.all([getClusterList(), window._env_.K8S_CLIENT ? { result: undefined } : getEnvironmentList()])
+            .then(([clusterRes, envResponse]) => {
+                const environments = envResponse.result || []
+                const clusterEnvironmentMap = environments.reduce((agg, curr) => {
+                    const newAgg = { ...agg }
+                    newAgg[curr.cluster_id] = newAgg[curr.cluster_id] || []
+                    newAgg[curr.cluster_id].push(curr)
+                    return newAgg
+                }, {})
+
+                let clustersList = clusterRes.result || []
+                clustersList = clustersList.map((cluster) => ({
+                    ...cluster,
+                    environments: clusterEnvironmentMap[cluster.id],
+                }))
+
+                clustersList = clustersList.sort((a, b) => sortCallback('cluster_name', a, b))
+
+                setClusters(clustersList)
+                setView(ViewType.FORM)
+            })
+            .catch((error) => {
+                showError(error)
+                setView(ViewType.ERROR)
+            })
+    }
+
+    const handleRedirectToClusterList = () => {
+        history.push(URLS.GLOBAL_CONFIG_CLUSTER)
+    }
+
+    useEffect(() => {
+        if (isSuperAdmin) {
+            initialize()
+        }
+    }, [isSuperAdmin])
+
+    if (!isSuperAdmin) {
+        return (
+            <div className="dc__align-reload-center">
+                <ErrorScreenNotAuthorized />
```

Comment:
The parsing can be done on line no 70. no need to do it again and again

Reply:
There's a side-effect above this method that shouldn't be called conditionally, lets align on this